pkg/deploy/deploy_rp.go

```diff
@@ -90,6 +91,17 @@ func (d *deployer) DeployRP(ctx context.Context) error {
 			continue
 		}
 		if err != nil {
+			// Try to remove failed VMSS so conflict doesn't occur during retry
+			// (when new VMSS with same name is deployed)
+			innerErr := d.rpRemoveFailedScaleset(ctx)
+			if innerErr != nil {
+				// Log instead of return so we don't mask original failure
+				// and so we can retry
+				d.log.Print(innerErr)
+			}
+			if i < 2 {
```

Comment:
This dependency on specific index, 1 and now 2, just.. it's wrong.  But it's not wrong enough for me to ding the PR I think.  Rather see an open loop and a set of conditions checked and exit if the conditions are met else continue with some cap on iterations.  Just had to write this down as e2e is still going and in case someone felt this needed addressed.  We have time.  But I'll approve.

Reply:
Yeah, using the specific `i` values is pretty nasty. There's got to be a better way.

I also have another concern: I'm not sure that retrying immediately after attempting to delete the failed VMSS will necessarily prevent the resource conflict. If it doesn't "go away" fast enough, there might be a race condition between deleting the resource and trying to recreate it. This is why I originally didn't retry, I just deleted the VMSS and returned the original error. I figured we could retry at the pipeline level, and just be spared the pain of manually deleting the VMSS.

Reply:
Maybe a condition if there's an existing VMSS and status is "Deleting" wait up to.. 10 minutes?

Reply:
Added the check and sleep